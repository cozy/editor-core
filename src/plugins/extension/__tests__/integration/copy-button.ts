import { _getCopyButtonTestSuite } from '../../../../../src/__tests__/integration/copy-button/_getCopyButtonTestSuite';
import fullWidthExtensionADF from '../visual-regression/__fixtures__/full-width-extension-inside-bodied-extension.adf.json';

_getCopyButtonTestSuite({
  nodeName: 'Extension',
  editorOptions: {
    allowExtension: true,
    defaultValue: fullWidthExtensionADF,
  },
  nodeSelector: '.extension-title',
});
