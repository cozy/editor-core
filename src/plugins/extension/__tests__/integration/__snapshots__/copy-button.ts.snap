// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy block with floating toolbar copy button 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "extensionKey": "bodied-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "default",
        "localId": "testId",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "extensionKey": "awesome:list",
            "extensionType": "com.atlassian.forge",
            "layout": "full-width",
            "localId": "testId",
            "parameters": Object {
              "items": Array [
                "a",
                "b",
                "c",
                "d",
              ],
            },
            "text": null,
          },
          "type": "extension",
        },
        Object {
          "content": Array [
            Object {
              "text": "Sample",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "bodiedExtension",
    },
    Object {
      "attrs": Object {
        "extensionKey": "bodied-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "default",
        "localId": "testId",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "extensionKey": "awesome:list",
            "extensionType": "com.atlassian.forge",
            "layout": "full-width",
            "localId": "testId",
            "parameters": Object {
              "items": Array [
                "a",
                "b",
                "c",
                "d",
              ],
            },
            "text": null,
          },
          "type": "extension",
        },
        Object {
          "content": Array [
            Object {
              "text": "Sample",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "bodiedExtension",
    },
  ],
  "type": "doc",
}
`;
